graphs/adapter-jgrapht/src/main/kotlin/com/anaplan/engineering/azuki/graphs/adapter/jgrapht/check/HasShortestPathCheck.kt

```diff
@@ -9,20 +9,28 @@ import org.jgrapht.graph.DefaultEdge
 
 class HasShortestPathCheck<V>(
     private val graphName: String,
-    private val from: V,
-    private val to: V,
-    private val shortestPath: List<V>
+    private val shortestPath: List<V>,
 ) : JGraphTCheck, GetShortestPathBehaviour() {
 
-    private val fullShortestPath by lazy {
-        listOf(from) + shortestPath + to
+    private val from = try {
```

Comment:
💡 Also, even if you wanted to keep the assertion in the `check` function, the appropriate way to handle these would have been as [lazy properties](https://kotlinlang.org/docs/delegated-properties.html#lazy-properties) rather than by catching an exception.